### PR TITLE
Lava is no longer a plating + all lava is smooth

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Lava is no longer a plating + all lava is smooth. I originally wanted to move it down past the floor type, but we need footsteps, so I'm keeping it there. Reverts #23480 code-wise, but keeps the fix functionally. Also removes the airless subtypes of lava. Removes an unused "lava" variable on floors

## Why It's Good For The Game
Prevents some weird oversights, and makes the code clean.

## Testing

Could still RCD over lava, "floor is lava" admin verb is not affected.

## Changelog
NPFC

### CODE REVIEW

File: code/game/turfs/simulated/floor/lava.dm [hunk 1]
```diff
@@ -1,8 +1,10 @@
-/turf/simulated/floor/plating/lava
+/turf/simulated/floor/lava
```
Comment (Sirryan2002) [inline code/game/turfs/simulated/floor/lava.dm:1]: please add some general documentation to this type. Explain why we relegate this to it's own floor subtype.

File: code/game/turfs/simulated/floor.dm [hunk 2]
```diff
@@ -19,7 +19,6 @@ GLOBAL_LIST_INIT(icons_to_ignore_at_floor_init, list("damaged1","damaged2","dama
 	var/icon_plating = "plating"
 	thermal_conductivity = 0.020
 	heat_capacity = 100000
-	var/lava = 0
```
Comment (Sirryan2002) [inline code/game/turfs/simulated/floor.dm:22]: lol

**Review by Sirryan2002 — APPROVED** [review]
obv needs to be deconflicted, but the code all checks out

**Review by S34NW — APPROVED** [review]
Deconflict for merge

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**